### PR TITLE
feat: support GitHub-flavored alerts

### LINKED ISSUE #3278: Support GitHub-style Alert?
### Is your feature request related to a problem? Please describe.

GitHub introduced a syntax to have alerts: https://github.com/orgs/community/discussions/16925

Basically, it's:

```
> [!NOTE]  
> Highlights information that users should take into account, even when skimming.

> [!TIP]
> Optional information to help a user be more successful.

> [!IMPORTANT]  
> Crucial information necessary for users to succeed.

> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.

> [!CAUTION]
> Negative potential consequences of an action.
```

> [!NOTE]  
> Highlights information that users should take into account, even when skimming.

> [!TIP]
> Optional information to help a user be more successful.

> [!IMPORTANT]  
> Crucial information necessary for users to succeed.

> [!WARNING]  
> Critical content demanding immediate user attention due to potential risks.

> [!CAUTION]
> Negative potential consequences of an action.

### Describe the solution you'd like

I think it would be nice for VitePress to support that built-in. So markdown files authored that way can have a closer experience when reading on GitHub.

I wrote a markdown-it plugin for that: https://github.com/antfu/markdown-it-github-alerts that VitePress might be able to use directly. (and also a short intro and preview on my site: https://antfu.me/posts/github-alerts)

I guess we don't need to align exactly the same as GitHub's styling. We could just render our existing alerts.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by antfu**
close #3278

Added three variants:
- `NOTE` (Same as `INFO`)
- `CAUTION` (Same as `DANGER`)
- `IMPORTANT` (with a new purple color).

Screenshot 2024-01-20 at 13 40 07

### CODE REVIEW

File: src/node/markdown/plugins/githubAlerts.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const markers = [
+    'TIP',
+    'NOTE',
+    'INFO',
+    'IMPORTANT',
+    'WARNING',
+    'CAUTION',
+    'DANGER'
+  ]
+  const matchCaseSensitive = true
```
Comment (brc-dd) [inline src/node/markdown/plugins/githubAlerts.ts:17]: Any specific reason for having this true? GitHub's implementation seems to be case insensitive.

Comment (antfu) [reply src/node/markdown/plugins/githubAlerts.ts:17]: I think it was case-sensitive, was copied from `markdown-it-github-alerts` https://github.com/antfu/markdown-it-github-alerts/commit/34f264da25042a40e50e0475bf1f5c9d392f339e

Let me update this

**Comment by antfu** [comment]
It can be another PR, but do you think we should add icons to those callouts and make it configurable like `infoLabel`?

**Review by brc-dd — APPROVED** [review]

**Comment by brc-dd** [comment]
> It can be another PR, but do you think we should add icons to those callouts and make it configurable like infoLabel?

In one PR Guillaume attempted adding it via CSS icons. We can do that probably and instead of making it configurable via config, better to just allow overriding the icon variables in custom CSS?

**Pull request merged**